### PR TITLE
Fix CI on travis and include Python 3.8-dev

### PR DESCRIPTION
**Pull request by bluebird75**
This fixes and adds the following points:
* python  3.7-dev is no longer working on Ubuntu Trusty image, use Ubuntu Xenial instead (see https://github.com/travis-ci/travis-ci/issues/9069)
* use official 3.7 release (with Ubuntu Xenial)
* provide a build for 3.8-dev allowing failures (with Ubuntu Xenial)
* use the more flexible matrix syntax
* run mypy for 3.6, 3.7, 3.8-dev versions (using a environement variable to trigger mypy)
* avoid latest mypy v0.620 which signals an error (to be fixed)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+          sudo: true
+          env: USE_MYPY=true
+    allow_failures:
+        - python: 3.8-dev
+  
 install:
-  - pip install -r requirements.txt
-  - if [[ $TRAVIS_PYTHON_VERSION == '3.6' ]]; then pip install -U mypy; fi
+    - pip install -r requirements.txt
+    - if [[ $USE_MYPY == true ]]; then pip install -U "mypy<0.620"; fi
```
Comment (gvanrossum) [inline .travis.yml]: Why install mypy older than 0.620?

Comment (gvanrossum) [reply .travis.yml]: (OK, I see your explanation -- but why not force 0.610? IIUC technically this would allow e.g. mypy 0.540 to be installed...)

Comment (bluebird75) [reply .travis.yml]: Indeed, that's much safer.

**Comment by gvanrossum** [comment]
Sorry, the 3.5.1 error is my fault. Can you fix it by changing the import to `from mypy_extensions import Type`? I think the other test failure is a Travis flake.

**Comment by bluebird75** [comment]
I tried but Type does not exist in mypy_extensions

File: pyannotate_runtime/collect_types.py [hunk 2]
```diff
@@ -49,7 +49,6 @@
     Set,
     Sized,
     Tuple,
-    Type,
     TypeVar,
     Union,
 )
```
Comment (gvanrossum) [inline pyannotate_runtime/collect_types.py]: To fix the test, add:
```py
MYPY = False
if MYPY:
    from typing import Type
```

**Comment by gvanrossum** [comment]
To fix the test, add:
```
MYPY = False
if MYPY:
    from typing import Type
```

**Comment by bluebird75** [comment]
Finally, this works !

**Comment by gvanrossum** [comment]
Thanks!

**Pull request merged**